### PR TITLE
[Rust Toolchain] Upgrade to rust version 1.7

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR updates the rust toolchain version from 1.66.0 (which is around 6 months old), to the latest version of rust, 1.7.0 (which is around 3 weeks old). All changes made in the PR were suggested by clippy. I've done basic due diligence to make sure no semantic changes were made, but would appreciate some additional eyes 😄 

### Test Plan
Existing test infrastructure.

### CODE REVIEW

**Review by zekun000 — APPROVED** [review]

File: consensus/src/round_manager_test.rs [hunk 1]
```diff
@@ -1331,7 +1331,7 @@ fn sync_on_partial_newer_sync_info() {
             .unwrap();
         // commit genesis and block 1
         for i in 0..2 {
-            let _ = node.commit_next_ordered(&[i]);
+            node.commit_next_ordered(&[i]).await;
```
Comment (msmouse) [inline consensus/src/round_manager_test.rs:1334]: huge win

File: consensus/consensus-types/src/block_test.rs [hunk 2]
```diff
@@ -108,9 +108,6 @@ fn test_block_relation() {
         genesis_block.id()
     );
     assert_eq!(next_block.payload(), Some(&payload));
-
-    let cloned_block = next_block.clone();
-    assert_eq!(cloned_block.round(), next_block.round());
```
Comment (msmouse) [inline consensus/consensus-types/src/block_test.rs:113]: 😂

**Review by msmouse — APPROVED** [review]

**Review by wrwg — APPROVED** [review]

**Comment by 0xjinn** [comment]
Nice! Can we update the changelog on the CLI to reflect the upgrade of rust version? `crates/aptos/CHANGELOG.md`

**Comment by JoshLind** [comment]
> Nice! Can we update the changelog on the CLI to reflect the upgrade of rust version? crates/aptos/CHANGELOG.md

SGTM 😄 Will send out a PR once this lands.

**Comment by sherry-x** [comment]
Hey Josh, seems we see invariant violation in forge for this PR https://cloud.us.humio.com/k8s/search?live=false&query=%24forgeLogs(validator_instance%3D*)%20%7C%0A%20%20%20%20%22k8s.namespace%22%20%3D%20%2Fforge%2F%20%2F%2F%20filters%20on%20namespace%20which%20contains%20validator%20logs%20%0A%7C%20message%20%3D%20%2Finvariant%20violation%2F%0A%7C%20groupby(%5B%23k8s.cluster%2C%20k8s.namespace%2C%20message%5D%2C%20function%3Dmax(%40timestamp))&start=2h&tz=America%2FLos_Angeles did you rebase to the latest of main? want to make sure it's not newly introduced by your PR

**Comment by JoshLind** [comment]
> Hey Josh, seems we see invariant violation in forge for this PR...

Interesting.. okay, thanks, let me rebase from main and retry 🤞

**Pull request merged**